### PR TITLE
add error flow AddToBalance

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

Add error handling if param is missing, onError toast notif
## Screenshots or screen recordings

CleanShot 2022-09-09 at 11 26 14
## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1946** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1946" target="_blank">Graphite</a> 
    * **PR #1947** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1947" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1947?utm_source=stack-comment).

File: src/components/Project/ProjectToolsDrawer/AddToProjectBalanceForm.tsx [hunk 1]
```diff
@@ -36,8 +37,17 @@ export function AddToProjectBalanceForm({
...
+        onError: e => {
+          setLoadingAddToBalance(false)
+          addToBalanceForm.resetFields()
+          emitErrorNotification(e.message)
+        },
       },
     )
+
+    if (!result) {
+      setLoadingAddToBalance(false)
```
Comment (tomquirk) [inline src/components/Project/ProjectToolsDrawer/AddToProjectBalanceForm.tsx:49]: **question**: I think this is also an error case right? Should we emit a generic error noti here?

File: src/hooks/v2/transactor/AddToBalanceTx.ts [hunk 2]
```diff
@@ -4,32 +4,46 @@ import { V2UserContext } from 'contexts/v2/userContext'
...
 
   const DEFAULT_MEMO = ''
 
   return ({ value }, txOpts) => {
-    if (!transactor || !projectId || !contracts?.JBETHPaymentTerminal) {
-      txOpts?.onDone?.()
-      return Promise.resolve(false)
-    }
+    try {
+      invariant(transactor && projectId && contracts?.JBETHPaymentTerminal)
```
Comment (tomquirk) [inline src/hooks/v2/transactor/AddToBalanceTx.ts:22]: Loving this more with every PR!

**Review by tomquirk — COMMENTED** [review]
@benschac thank you!!! One comment

File: src/components/Project/ProjectToolsDrawer/AddToProjectBalanceForm.tsx (hunk 1 above)
Comment (benschac) [reply src/components/Project/ProjectToolsDrawer/AddToProjectBalanceForm.tsx:49]: CleanShot 2022-09-16 at 12 26 59

Comment (benschac) [reply src/components/Project/ProjectToolsDrawer/AddToProjectBalanceForm.tsx:49]: I don't think so. The emit error is getting called in the `onError` call.

**Comment by tomquirk** [comment]
Awsome, lgtm!

**Review by tomquirk — APPROVED** [review]

**Pull request merged**